### PR TITLE
Receipt: Add payment proof

### LINKED ISSUE #4685: `Receipt` instead of BOLT11 show preimage
Currently on the receipt page we show bolt11 as a proof of payment ,which not only is irrelevant information for anyone willing to check if the invoice is paid, but due to it's length makes receipts oddly long. I think we do the same for payment requests.

I propose we add preimage/hash as an actual veritable proof of payment. 

With this embedded, I don't think we need `Show the payment list in the public receipt page` in settings, as any recepit should have a proof of payment, that's the point of a receipt. However the later can be split up in a different issue.

https://user-images.githubusercontent.com/36959754/220588069-da19d907-3107-4e6b-94f1-c3864f304608.MOV

### PR DESCRIPTION
**Pull request by dennisreimann**
We now display a payment proff on the receipt page:

- Onchain: Transaction Id
- Lightning: Preimage

Closes #4685.

grafik

### CODE REVIEW

**Comment by dennisreimann** [comment]
TODO:

- [ ] Onchain: Remove duplicate Payment Proof
- [ ] Offchain: Remove Transaction Id

**Comment by dennisreimann** [comment]
Improved the view and added a view component for displaying truncated text — it's configurable, displays things properly when printed and optionally shows a copy button. We can re-use it elsewhere once this is merged.

grafik

**Comment by dstrukt** [comment]
Only thought w.r.t to the TX id's ( i ) is that it might not be immediately obvious it's a click away link .. wondering if we'll want to use a more common icon for that .. thoughts?

Otherwise, looks much better and cleaner .. going to explore some more options tomorrow when I'm in Figma.

**Comment by dennisreimann** [comment]
I reused what we have on the labels, but am willing to adapt both if you find a better solution :)

**Comment by NicolasDorier** [comment]
Something is strange.

The `Transaction Id` in on-chain term is only the tx hash, here it is the outpoint. (`txid-idx`)
The `Proof` should be the outpoint.

Here they are the same.

File: BTCPayServer/Components/TruncateCenter/TruncateCenter.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace BTCPayServer.Components.TruncateCenter
+{
+    public class TruncateCenter : ViewComponent
```
Comment (NicolasDorier) [inline BTCPayServer/Components/TruncateCenter/TruncateCenter.cs:5]: small comment to understand what is that?

File: BTCPayServer/Services/Invoices/InvoiceEntity.cs [hunk 2]
```diff
@@ -1330,5 +1330,6 @@ public interface CryptoPaymentData
 
         PaymentType GetPaymentType();
         string GetDestination();
+        string GetPaymentProof();
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Invoices/InvoiceEntity.cs:1333]: Not sure it's good idea to make this interface bigger. If anything, we should probably break it down later.
Anyway, leave it like that for now, I may change your PR later.

File: BTCPayServer/Components/TruncateCenter/TruncateCenter.cs (hunk 1 above)
Comment (dennisreimann) [reply BTCPayServer/Components/TruncateCenter/TruncateCenter.cs]: Added in [83eadc2](https://github.com/btcpayserver/btcpayserver/pull/4782/commits/83eadc2bcb9b6014f9f073c6376e7a3d765b09a3).

**Comment by dennisreimann** [comment]
@NicolasDorier I've now removed the Transaction ID, as we use `GetPaymentId()` in several places and I didn't want to touch it, because it most likely assumed the outpoint.

Also this let's us display everything consistent and independent of the payment method. Added the link to the payment proof, so that it is also still present. 

grafik

You can take over now :)

**Comment by NicolasDorier** [comment]
I removed interface and just hard coded for LN and bitcoin payments.

This is less flexible, but we will soon refactor all of this anyway.

**Pull request merged**